File: meme_inator_front/lib/features/profiles/domain/usecases/get_profile_img_urls_usecase.dart
import 'package:meme_inator_front/core/results.dart';
import 'package:meme_inator_front/features/profiles/domain/interfaces/iprofile_repository.dart';
import 'package:uuid/uuid.dart';


class GetProfileImageUrlsUsecase {
  final IProfileRepository profileRepo;

  GetProfileImageUrlsUsecase(this.profileRepo);

  Future<Result<Map<String, String>>> execute({
    required UuidValue userId,
  }) async {
    throw UnimplementedError('GetProfileImageUrlsUsecase.execute not implemented');
  }
}

File: meme_inator_back/apps/profiles/application/usecases/get_profile_img_urls_usecase.py
from typing import Dict
from uuid import UUID
from apps.profiles.domain.usecases.iget_profile_img_urls_usecase import IGetProfileImageUrlsUsecase
from apps.profiles.domain.irepositories.i_profile_repository import IProfileRepository
from core.results import Result

class GetProfileImageUrlsUsecase(IGetProfileImageUrlsUsecase):
    def __init__(self, profile_repo: IProfileRepository):
        self.profile_repo = profile_repo

    def execute(self, user_id: UUID) -> Result[Dict[str, str]]:
        return self.profile_repo.get_profile_image_urls(user_id)